src/Symfony/Component/Console/Helper/TableStyle.php

```diff
@@ -132,6 +139,127 @@ public function getCrossingChar()
         return $this->crossingChar;
     }
 
+    /**
+     * Sets crossing characters.
+     *
+     * Example:
+     * <code>
+     * 1---------------2-----------------------2------------------3
+     * | ISBN          | Title                 | Author           |
+     * 8---------------0-----------------------0------------------4
+     * | 99921-58-10-7 | Divine Comedy         | Dante Alighieri  |
+     * | 9971-5-0210-0 | A Tale of Two Cities  | Charles Dickens  |
+     * | 960-425-059-0 | The Lord of the Rings | J. R. R. Tolkien |
+     * 7---------------6-----------------------6------------------5
+     * </code>
+     *
+     * @param string      $cross       Crossing char (see #0 of example)
+     * @param string|null $topLeft     Top left char (see #1 of example), equal to $cross if null
+     * @param string|null $topMid      Top mid char (see #2 of example), equal to $cross if null
+     * @param string|null $topRight    Top right char (see #3 of example), equal to $cross if null
+     * @param string|null $midRight    Mid right char (see #4 of example), equal to $cross if null
+     * @param string|null $bottomRight Bottom right char (see #5 of example), equal to $cross if null
+     * @param string|null $bottomMid   Bottom mid char (see #6 of example), equal to $cross if null
+     * @param string|null $bottomLeft  Bottom left char (see #7 of example), equal to $cross if null
+     * @param string|null $midLeft     Mid left char (see #8 of example), equal to $cross if null
+     *
+     * @return $this
+     */
+    public function setCrossingChars(string $cross, string $topLeft = null, string $topMid = null, string $topRight = null, string $midRight = null, string $bottomRight = null, string $bottomMid = null, string $bottomLeft = null, string $midLeft = null): self
```

Comment:
imo I would rather use `?string` as @javiereguiluz said (or even not nullable `string`) and not take `crossingChar`. This methid should be called with `$this->setCrossingChars($cross, $cross, ...);` (and thus slowly deprecate `crossingChar`)

Reply:
Not sure to understand why use nullable string here... Could you to be more accurate please?

Reply:
Because `?string $topLeft` looks better than `string $topLeft = null` and the `?string` feature was introduced in modern PHP versions precisely to avoid doing `string ... = null`

Reply:
Mmh I'm not sure that `?string $topLeft` replaces ` string $topLeft = null`. `?string` simply allows `string` and `null` value, but it doesn't set a null default value. 
```php
function f(?string $p) {}
f(); // Uncaught Error: Too few arguments to function f(), 0 passed in...
```

Reply:
More info here: http://php.net/manual/en/migration71.new-features.php

Reply:
mmmmm ... you are so right!!! I was too focused on the code and not on the feature. You allow to not pass this argument, so `$string` would be wrong indeed! I'm really sorry for all this noise 😊

Reply:
(The first time, I misunderstood as you and It's only when I tried to use `?string` in this PR that I understood its real meaning :smile:)

Reply:
I didn't misunderstood AFAIC. I think that not giving these arguments a default should be the way to go, and remove the `$crossingChar` argument. I think the whole point of this PR should be deprecated `$crossingChar` and split it into 6 parts.

So basically : `public function setCrossingChars(?string $topLeft, ?string $topMid, ?string $topRight, ?string $midRight, ?string $bottomRight, ?string $bottomMid, ?string $bottomLeft, ?string $midLeft): self` or even to push it further `public function setCrossingChars(string $topLeft, ?string $topMid, string $topRight, string $midRight, string $bottomRight, string $bottomMid, string $bottomLeft, string $midLeft): self`.

Why ? Once again, because this is the goal of this PR imo. And having these values set to null doesn't really have any sense, as you *should* give these values a value, even a default one (all the asciis symbols, or whatever pleases use, not an absence of a value).

The deprecation of the `$crossingChar` would then lead the `setCrossingChar(string $crossingChar)` would then just to the following :

```php
<?php

// ...
private $crossingChar; // @deprecated
private $crossingChars = [....]; // default values

public function getCrossingChar(): string
{
    // trigger error message, this is deprecated
    return $this->deprecated;
}

public function setCrossingChar(string $crossingChar): self
{
    // trigger error message, this is deprecated
    $this->crossingChar = $crossingChar;
    $this->setCrossingChars($crossingChar, $crossingChar, ...);

    return $this;
}

public function setCrossingChars(string $topLeft, ...): self
{
    $this->topLeft = ...;
    ...
}
```

Do you see where I'm getting at ?

Reply:
I'm sorry, I'm still not sure to understand your point. `$crossingChar` is absolutely necessary for rendering a table, and it can't be replaced by `$topLeft` (`┼` != `┌`). Deprecate it is impossible.

Reply:
I'm still not sure to understand your point. `$crossingChar` is absolutely necessary for rendering a table, and it can't be replaced by `$topLeft` (`┼` != `┌`). Remove it is impossible.

Reply:
Hum, didn't see the `0`, so my bad for the removal. But I still think that allowing a `null` value for the rest is not really good, and you *should* specify a value for all these other edges.

Reply:
I mean, you CAN use the `crossingChar` for ALL the edges, but it shouldn't be implied if no value is given. Hence the "not allowing null values", so nothing too implicit.

Reply:
> allowing a null value for the rest is not really good

I think that the first param `$crossingChar` can be used as _generic_ crossing char:

```php
$style->setCrossingChars('+');
// vs
$style->setCrossingChars('+', '+', '+', '+', '+', '+', '+', '+', '+');
```

Reply:
i'd add this method above `getCrossingChar`

Reply:
@ro0NL Moved.